src/hooks/usePublicMessages.js

```diff
@@ -78,19 +80,27 @@ async function fetchPublicMessages(infoPoint) {
   const getCurrentMessagesResponse = await fetch(new URL('PublicMessages/GetCurrentMessages', infoPoint));
   const getCurrentMessagesJSON = await getCurrentMessagesResponse.json();
   getCurrentMessagesJSON.forEach((publicMessage) => {
+    let sortOrder;
+    const routes = [];
     publicMessage['Routes'].forEach((routeId) => {
-      const route = routesById[routeId] || {};
-      publicMessages.push({
-        key: [publicMessage['MessageId'], route['RouteId']].filter((presence) => (presence)).join('-'),
-        message: publicMessage['Message'],
-        routeAbbreviation: route['RouteAbbreviation'] || null,
-        routeColor: route['Color'] || null,
-        routeTextColor: route['TextColor'] || null,
-        routeSortOrder: route['SortOrder'] || null,
+      routes.push({
+        id: routeId,
+        abbreviation: routesById[routeId]['RouteAbbreviation'] || 'ALL',
+        color: routesById[routeId]['Color'] || null,
+        textColor: routesById[routeId]['TextColor'] || null,
+        sortOrder: routesById[routeId]['SortOrder'] || null,
       });
+      if (routesById[routeId]['SortOrder'] && routesById[routeId]['SortOrder'] < sortOrder) {
+        sortOrder = routesById[routeId]['SortOrder'];
+      }
+    });
+    publicMessages.push({
+      key: publicMessage['MessageId'],
+      message: publicMessage['Message'],
+      routes: routes.sort(compareRoutes),
```

Comment:
I think I don't mind if sorting occurs here rather than in the hook, but I think I do mind if we do route sorting vs general sorting in two different places (i.e. I think we should either move the public messages sort here, or move this to the hook).